content/api/commands/origin.md

```diff
@@ -0,0 +1,467 @@
+test, verify the new version is displayed on a different URL in the third, and
+logout in the fourth. Here's a simplified example of such a test strategy.
+
+<Badge type="danger">Before</Badge> Multiple small tests against different
+origins
+
+```js
+it('logs in', () => {
+  cy.visit('https"//supersecurelogons.com')
+  cy.get('input#password').type('Password123!')
+  cy.get('button#submit').click()
+})
+
+it('updates the content', () => {
+  cy.get('#current-user').contains('logged in')
+  cy.get('button#edit-1').click()
+  cy.get('input#title').type('Updated title')
+  cy.get('button#submit').click()
+  cy.get('.toast').type('Changes saved!')
+})
+
+it('validates the change', () => {
+  cy.visit('/items/1')
+  cy.get('h1').contains('Updated title')
+})
+```
+
+After switching on `experimentalSessionAndOrigin`, this flow would need to be
+contained within a single test. While this practice has always been
+[discouraged](/guides/references/best-practices#Having-tests-rely-on-the-state-of-previous-tests)
+we know some users have historically written tests this way, often to get around
+the same-origin restrictions. But with `cy.origin()` you no longer need these
+kind of brittle hacks, as your multi-origin logic can all reside in a single
+test, like the following.
+
+<Badge type="success">After</Badge> One big test using `cy.origin()`
+
+```js
+it('securely edits content', () => {
+  cy.origin('supersecurelogons.com', () => {
+    cy.visit('https"//supersecurelogons.com')
+    cy.get('input#password').type('Password123!')
+    cy.get('button#submit').click()
+  })
+
+  cy.origin('mycms.com', () => {
+    cy.url().should('contain', 'cms')
+    cy.get('#current-user').contains('logged in')
+    cy.get('button#edit-1').click()
+    cy.get('input#title').type('Updated title')
+    cy.get('button#submit').click()
+    cy.get('.toast').type('Changes saved!')
+  })
+
+  cy.visit('/items/1')
+  cy.get('h1').contains('Updated title')
+})
+```
+
+Always remember,
+[Cypress tests are not unit tests](https://docs.cypress.io/guides/references/best-practices#Creating-tiny-tests-with-a-single-assertion).
+
+### Serialization
+
+When entering a `cy.origin()` block, the test-runner injects the Cypress
+runtime, with all your configurations settings, into the requested origin, and
+sets up bidirectional communication with that instance. This coordination model
+requires that any data sent from one instance to another be
+[serialized](https://developer.mozilla.org/en-US/docs/Glossary/Serialization)
+for transmission. It is very important to understand that variables **inside**
+the callback are not shared with the scope **outside** the callback. For example
+this will not work:
+
+```js
+const foo = 1
+cy.origin('somesite.com', () => {
+  cy.visit('/')
+  // This line will throw a ReferenceError because `foo` is not defined in the
+  // scope of the callback
+  cy.get('input').type(foo)
+})
+```
```

Comment:
Usually when providing examples that don't work we need big ❗ icons. People...will.....copy....this...code! And not read the actual text that this is a non-working example. 😅  Example: https://docs.cypress.io/guides/core-concepts/introduction-to-cypress#Commands-Are-Asynchronous